scripts/schemas/soc-schema.yml

```diff
@@ -70,3 +70,56 @@ mapping:
     required: false
     type: str
     desc: Free form comment with extra information regarding the SoC.
+  run_once:
+    type: seq
+    desc: |
+          Allows for restricting west flash commands when using sysbuild to run once per given
+          grouping of boards. This is to allow for future image program cycles to not erase the
+          flash of a device which has just been programmed by another image.
+    sequence:
+      - type: map
+        mapping:
+          regex;(.*):
+            required: true
+            type: seq
+            desc: |
+                  A list of commands which should be limited to running once per invocation of
+                  west flash for a set of boards
+            sequence:
+              - type: map
+                mapping:
+                  boards:
```

Comment:
soc.yml should not refer to boards.

Please change this to socs, so that soc.yml just define the socs to which the command belongs, then the logic in Python must map this to the relevant boards.

Just like what we do with CPU clusters. The soc.yml doesn't define the boards to which the CPU clusters applies, instead the board inherits the CPU clusters defined by the SoC the board is using.

Similar, let the SoC define the commands and the board using that SoC can then inherit the commands.

Reply:
This would create a mass of redundant extra code to perform lookups without any actual gain

Reply:
> Similar, let the SoC define the commands and the board using that SoC can then inherit the commands.

As discussed and decided at architecture working group this is purposely not the cause, a board will override the whole soc configuration